machine.js

```diff
@@ -154,6 +155,7 @@ function transitionTo(service, machine, fromEvent, candidates) {
         original: { value: original }
       });
 
+      logger(machine, to, service.context, context, fromEvent);
```

Comment:
What I mean is something like this:

```js
if(d._log) d._log(machine, to, service.context, context, fromEvent);
```

This prevents all of the passing a logger around and adding extra params to interpret.

Reply:
`d._log` is probably not right here. It's logging one specific thing, entering a new state. So probably `d._enter` is better.

Reply:
I added a default value to the logger parameter. See line 185. So the logging function is always set and I can omit this condition and the default signature of the interpret function stays the same. This is also reflected in the type script types.

Reply:
What are your thoughts on the above idea?

Reply:
The log function is always set to the emptyFN, so the condition is always true. Renaming it to `_log` would mark it as private IMO. Naming it to `enter` could be a better solution. Then it would be a more general callback. What do you think of `onEnter`?

Reply:
Yeah, the _ is private because it's calling out to the debug module.  `d._enter` seems right for this case. The debug module would then handle the API for logging. Maybe the user provides their logger using some method like this:

```js
import debug from 'robot3/debug';

debug.onEnter(info => {
  console.log(info);
});
```

Or something like this.

Reply:
Ah okay now I know what you are meaning. Seems a valid solution, I'll change the PR.